packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/helpers.ts

```diff
@@ -227,7 +227,9 @@ export const createBulkIndexOperationTuple = (
     {
       index: {
         _id: doc._id,
-        ...(typeIndexMap[doc._source.type] && { _index: typeIndexMap[doc._source.type] }),
+        ...(typeIndexMap[doc._source.type] && {
+          _index: typeIndexMap[doc._source.type] + '_alias',
+        }),
```

Comment:
`createBulkIndexOperationTuple` is a fairly low-level utility. Are we sure it's fine to hard-code this here? Isn't it used in scenarios where we're not targeting the temp index (alias)?

`createBulkIndexOperationTuple` is used by `createBatches` which is used both for v2 and zdt. 

<img width="562" alt="Screenshot 2023-05-30 at 08 40 34" src="https://github.com/elastic/kibana/assets/1532934/298bb6f1-3448-4d0b-95d0-95a0e78231dc">

And even for v2, I think this change is wrong for reindex-into-the-same-index scenario?

Reply:
Yes I think you're right, I would expect it to fail here https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts#L1291

Reply:
Yeah, we do have integration tess for ZDT too, I'm not sure to get why those are not failing with the PR's changes to be honest